### DISCUSSION TITLE
Make opposite-shift mode customizable

### DISCUSSION THREAD

**Discussion by bsamseth** [monkeytypegame/monkeytype#1493] [Ideas] [answered]
(A potential solution to #1199 and any issues like it)

The current implementation is fine for anyone on an (English) qwerty layout (i.e. almost everybody but a few), but it understandably won't work for non-standard layouts. And even with the standard layout, there are some keys which people can have differing opinions about what is the _correct_ shift key. 

As accounting for all kinds of layouts and subjective opinions won't really be feasible, I suggest that we make it configurable. Meaning, in addition to the on/off switch, we make a setting that allows the user to say which keys should go on which side, left or right shift. I would have it defaulted to the current values for qwerty, but editable.

I propose doing something like this for how you set which keys go where (obviously to be made much smarter+prettier, just to give the idea): https://codepen.io/bsamseth/full/wvJpwVr

I'd be happy to make a PR with this if it sounds like something that would be accepted, but I wanted to hear what your thought are before spending any time on it.

As always, hope you are well, and thanks for making a great site!

**Comment by Miodec** [comment, marked as answer]
Yeah, could add the left and right input fields, and then if they are empty just revert to the default qwerty mode. Just like custom layout fluid
